roles/ara_api/tasks/install/podman.yaml

```diff
@@ -0,0 +1,38 @@
+---
+- name: Detecting existing PyPI installation
+  ansible.builtin.stat:
+    path: "{{ ara_api_venv_path }}"
+  register: existing_pypi_install
+
+- name: Notify about existing PyPI installation
+  ansible.builtin.debug:
+    msg: |
+      You seem to have ARA-API installed via PyPI in the past, you might
+      want to clean up that installation and migrate your data
+  when: existing_pypi_install['stat']['exists']
+
+- name: Override file locations with path in container
+  ansible.builtin.set_fact:
+    ara_api_database_name: "/opt/ara/ansible.sqlite"
+    ara_api_log_dir: "/opt/ara/logs"
+    ara_api_settings: "{{ ara_api_root_dir }}/settings.yaml"
+
+- name: Ensure ARA API container
```

Comment:
There is kind of a gotcha here.

I think it is succeeding in starting the container when running from an unprivileged user (since podman can start rootless no problem) but then it tries to set up the systemd unit for it and fails due to lack of privileges.

If we set ``become: true`` on the task that starts the container, I would presume that it'd start it as root and not as the unprivileged user. It would work, but wouldn't be the same thing. If we go that route it would be worthy of being mentioned as a caveat since the pip/source installs supports running as unprivileged.

Note that there would also be a failure on the following task (service) since it doesn't have ``become: true``.

Reply:
I propose that we don't use the ``generate_systemd`` argument of the ``podman_container`` module and instead use podman_generate_systemd: https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_generate_systemd_module.html#examples

```yaml
# Example of creating a container and integrate it into systemd
- name: A postgres container must exist, stopped
  containers.podman.podman_container:
    name: postgres_local
    image: docker.io/library/postgres:latest
    state: stopped

- name: Systemd unit files for postgres container must exist
  containers.podman.podman_generate_systemd:
    name: postgres_local
    dest: ~/.config/systemd/user/

- name: Postgres container must be started and enabled on systemd
  ansible.builtin.systemd:
    name: container-postgres_local
    scope: user
    daemon_reload: true
    state: started
    enabled: true
```

This way we don't need to elevate privileges for the task that starts the container so the ara API container can remain rootless unless the role is run as root.

When running as root it'd look more like this (from the examples):
```
- become: true
  block:
    - name: Generate systemd unit file for postgres container
      containers.podman.podman_generate_systemd:
        name: postgres
        new: true
        no_header: true
        dest: /etc/systemd/system

    - name: Ensure postgres container is started and enabled
      ansible.builtin.systemd:
        name: container-postgres
        daemon_reload: true
        state: started
        enabled: true
```

Reply:
I have tried to do this but it's still WIP: https://github.com/dmsimard/recordsansible.ara/commit/8fa017ef3a42d0c61e76c6dd4e2b3c8fc9345eb3

In truth I am running into unrelated issues amongst which debian 12 is not supported so it would break when I wanted to test this with it: https://github.com/ansible-community/ara-collection/pull/68

I will revive that PR.